core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.ts

```diff
@@ -0,0 +1,127 @@
+    var submittedAnswerInfoCount = 0;
+    var thresholdProbabilityIndex = null;
+    var randomProbabilityIndex = null;
+    var expId = null;
+    var exploration = null;
+    var state = null;
+    var stateName = null;
+    var interactionId = null;
+    var defaultOutcome = null;
+    var currentAnswer = null;
+    var currentInteractionRulesService = null;
+    var canAskLearnerForAnswerInfo = false;
+    var visitedStates = [];
+
+
+    var getRandomProbabilityIndex = function() {
+      var min = 0;
+      var max = 100;
+      return (Math.floor(Math.random() * (max - min + 1) ) + min) / 100;
+    };
+
+    return {
+      initLearnerForAnswerInfo: function(
+          answer, interactionRulesService) {
+        currentAnswer = answer;
+        currentInteractionRulesService = interactionRulesService;
+        expId = ExplorationEngineService.getExplorationId();
+        exploration = ExplorationEngineService.getExploration();
+        stateName = PlayerTranscriptService.getLastStateName();
+        state = exploration.getState(stateName);
+        interactionId = state.interaction.id;
+        defaultOutcome = state.interaction.defaultOutcome;
+
+        if (submittedAnswerInfoCount === 2) {
+          return;
+        }
+
+        if (!state.solicitAnswerDetails) {
+          return;
+        }
+
+        if (INTERACTION_IDS_WITHOUT_ANSWER_DETAILS.indexOf(
+          interactionId) !== -1) {
+          return;
+        }
+
+        if (visitedStates.indexOf(stateName) !== -1) {
+          return;
+        }
+
+        visitedStates.push(stateName);
+
+        var classificationResult = (
+          AnswerClassificationService.getMatchingClassificationResult(
+            stateName, state.interaction, answer,
+            interactionRulesService));
+        var outcome = classificationResult.outcome;
+
+        randomProbabilityIndex = getRandomProbabilityIndex();
+        /* eslint-disable dot-notation */
+        if (outcome === defaultOutcome) {
+          thresholdProbabilityIndex = PROBABILITY_INDEXES['type_a'];
+        } else if (outcome.labelledAsCorrect) {
+          thresholdProbabilityIndex = PROBABILITY_INDEXES['type_b'];
+        } else {
+          thresholdProbabilityIndex = PROBABILITY_INDEXES['type_c'];
+        }
+        /* eslint-enable dot-notation */
+        canAskLearnerForAnswerInfo = (
+          randomProbabilityIndex <= thresholdProbabilityIndex);
+        return canAskLearnerForAnswerInfo;
+      },
+      resetSubmittedAnswerInfoCount: function() {
+        submittedAnswerInfoCount = 0;
+      },
+      recordLearnerAnswerInfo: function(answerDetails) {
+        // Replace this comment with LearnerBackendApiService function,
+        // once the other PR gets in.
+        submittedAnswerInfoCount++;
+        canAskLearnerForAnswerInfo = false;
+      },
+      canAskLearnerForAnswerInfo: function() {
+        return canAskLearnerForAnswerInfo;
+      },
+      getCurrentAnswer: function() {
```

Comment:
Is this for the interaction answer or answerInfo?

Reply:
Its the interaction answer